mcs/class/System.Core/System.IO.MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -93,19 +93,38 @@ internal static void Map (IntPtr handle, long offset, ref long size, MemoryMappe
 			}
 		}
 
+		internal static void CheckString (string s)
+		{
+			// Underlying implementation does not handle embedded nuls.
+			if (s == null || s.IndexOf((char)0) != -1)
+				return;
+			throw new ArgumentException ($"String \"{s}\" contains embedded nul.");
```

Comment:
This changes the behavior of the API. Why is it needed?

Reply:
I agree it is separate, maybe a separate PR? The preexisting behavior was faulty -- if presented with string with embedded nuls it would silently truncate. Trying to open foo\0bar should fail and not silently open foo.